chain/beacon/store.go

```diff
@@ -198,35 +205,48 @@ func (c *callbackStore) Put(ctx context.Context, b *chain.Beacon) error {
 }
 
 // AddCallback registers a function to call
-func (c *callbackStore) AddCallback(id string, fn func(*chain.Beacon)) {
+func (c *callbackStore) AddCallback(id string, fn CallbackFunc) {
 	c.Lock()
 	defer c.Unlock()
+	if _, exists := c.newJob[id]; exists {
+		c.newJob[id] <- cbPair{
+			cb:    c.callbacks[id],
+			b:     nil,
+			close: true, // Signal we close this job
+		}
+		close(c.newJob[id])
```

Comment:
if the channel contents aren't consumed won't this cause messages to be missed?
I tried it something similar [here](https://go.dev/play/p/4Ez1eikk1gs) and it deadlocks, but perhaps I'm missing something

Reply:
ahhh okay I think I get it - this is for existing callbacks that might already be in flight?